test/test-cases/ssr-hello-world/ssr-hello-world.test.js

```diff
@@ -0,0 +1,72 @@
+const { promisify } = require('es6-promisify');
+const rimrafAsync = promisify(require('rimraf'));
+const { exec } = require('child-process-promise');
+const spawn = require('../../utils/gracefulSpawn');
+const waitOnAsync = promisify(require('wait-on'));
+const fetch = require('node-fetch');
+const skuConfig = require('./sku.config');
+
+const backendUrl = `http://localhost:${skuConfig.port.backend}`;
+const clientJsUrl = `http://localhost:${skuConfig.port.client}/main.js`;
+
+describe('ssr-hello-world', () => {
+  describe('start', () => {
+    let server;
+
+    beforeAll(async () => {
+      await rimrafAsync(`${__dirname}/dist/`);
+
+      server = spawn('node', [`${__dirname}/../../../scripts/start-ssr`], {
+        stdio: 'inherit',
+        cwd: __dirname
+      });
+      
+      await waitOnAsync({
+        resources: [
+          backendUrl.replace(/^http/, 'http-get'),
+          clientJsUrl.replace(/^http/, 'http-get')
+        ],
+        headers: {
+          accept: 'text/html, application/javascript'
+        },
+        timeout: 20000
```

Comment:
There are a few timeouts littered around, should they be uniform or in a config? Or are they deliberately set separately?

Reply:
Yeah, good idea, these should be centralised somewhere. I'd probably solve this at a higher level and make a `waitForUrls` util.

Reply:
Just did `npm run test` 8 times, 2 of them failed on ssr-hello-world.test.js because of the timeout error(took 24-ish seconds), the other 6 tests finished in 11 seconds. Do you think we can increase it to  higher value `25000` or pass a cb in waitOn to handle the error?

```
 FAIL  test/test-cases/ssr-hello-world/ssr-hello-world.test.js (24.381s)
```
```
Timeout
      at node_modules/wait-on/lib/wait-on.js:92:10
```
```
FetchError: request to http://localhost:8001/ failed, reason: connect ECONNREFUSED 127.0.0.1:8001
```